status/views.py

```diff
@@ -130,22 +120,13 @@ def view_code(request, sid):
             f.close()
             codesubmitform = CodeSubmitForm(
                 initial={'code': code, 'pid': submission.problem.id})
-            return render(request, 'users/submit.html', {'form': codesubmitform})
+            return render_index(request, 'users/submit.html', {'form': codesubmitform})
         else:
             logger.warning('User %s attempt to view detail of SID %s' % (request.user, sid))
-            return render(
-                request,
-                'index/500.html',
-                {'error_message': 'You don\'t have permission to view detail of SID %s' % sid})
+            raise PermissionDenied('You don\'t have permission to view detail of SID %s' % sid)
     except Submission.DoesNotExist:
         logger.warning('SID %s Not Found!' % sid)
-        return render(
-            request,
-            'index/500.html',
-            {'error_message': 'SID %s Not Found!' % sid})
+        raise Http404
     except IOError:
         logger.warning('File %s.cpp Not Found!' % sid)
-        return render(
-            request,
-            'index/500.html',
-            {'error_message': 'File of SID %s Not Found!' % sid})
+        raise Http404
```

Comment:
I prefer to redirect to 500 error page, with error message "source code doesn't exist".